tests/test_pytypes.py

```diff
@@ -1048,3 +1048,46 @@ def test_typevar(doc):
     assert doc(m.annotate_listT_to_T) == "annotate_listT_to_T(arg0: list[T]) -> T"
 
     assert doc(m.annotate_object_to_T) == "annotate_object_to_T(arg0: object) -> T"
+
+
+@pytest.mark.skipif(
+    not m.defined_PYBIND11_HAS_RANGES, reason="C++20 <ranges> not available."
+)
+def test_ranges(capture):
```

Comment:
The `capture` is best avoided. See my other comment.
This test will be much more concise if you use `assert ... == [2, 3, 4]`.